sdk/src/wallet/operations/syncing/mod.rs

```diff
@@ -94,21 +94,40 @@ where
     async fn sync_internal(&self, options: &SyncOptions) -> crate::wallet::Result<()> {
         log::debug!("[SYNC] sync_internal");
 
+        let wallet_address = self.address().await;
+
         let wallet_address_with_unspent_outputs = AddressWithUnspentOutputs {
-            address: self.address().await,
+            address: wallet_address.clone(),
             output_ids: self.ledger().await.unspent_outputs().keys().copied().collect(),
         };
 
-        let address_to_sync = vec![
-            wallet_address_with_unspent_outputs,
-            AddressWithUnspentOutputs {
-                address: self.implicit_account_creation_address().await?,
-                output_ids: vec![],
-            },
-        ];
+        let mut addresses_to_sync = vec![wallet_address_with_unspent_outputs];
+
+        if options.sync_implicit_accounts {
+            if let Ok(implicit_account_creation_address) = self.implicit_account_creation_address().await {
+                addresses_to_sync.push(AddressWithUnspentOutputs {
+                    address: implicit_account_creation_address,
+                    output_ids: self
+                        .ledger()
+                        .await
+                        .unspent_outputs
+                        .values()
+                        .filter_map(|output_data| {
+                            if output_data.output.is_implicit_account()
+                                && output_data.output.as_basic().address() == wallet_address.inner()
```

Comment:
Shouldn't it be `implicit_account_creation_address`?

Reply:
Fixed